### DISCUSSION TITLE
Files from Audiomoth not being recognised

### DISCUSSION THREAD

**Discussion by dsbennett2023** [kitzeslab/opensoundscape#900] [Q&A] [answered]
Good evening!
Im sorry if Ive made a mistake, still a beginner to python and this software. 

Im attempting to train a CNN in Google Colab, and have successfully completed the whole process with your tutorial files. 

Ive now attempted to create a CNN using my own files and species. However, when I load in four example files I get this error:
""LibsndfileError                           Traceback (most recent call last)
[/usr/local/lib/python3.10/dist-packages/librosa/core/audio.py](https://localhost:8080/#) in get_duration(y, sr, S, n_fft, hop_length, center, path, filename)
    794         try:
--> 795             return sf.info(path).duration  # type: ignore
    796         except sf.SoundFileRuntimeError:

11 frames
[/usr/local/lib/python3.10/dist-packages/soundfile.py](https://localhost:8080/#) in info(file, verbose)
    466     """
--> 467     return _SoundFileInfo(file, verbose)
    468 

[/usr/local/lib/python3.10/dist-packages/soundfile.py](https://localhost:8080/#) in __init__(self, file, verbose)
    411         self.verbose = verbose
--> 412         with SoundFile(file) as f:
    413             self.name = f.name

[/usr/local/lib/python3.10/dist-packages/soundfile.py](https://localhost:8080/#) in __init__(self, file, mode, samplerate, channels, subtype, endian, format, closefd)
    657                                          format, subtype, endian)
--> 658         self._file = self._open(file, mode_int, closefd)
    659         if set(mode).issuperset('r+') and self.seekable():

[/usr/local/lib/python3.10/dist-packages/soundfile.py](https://localhost:8080/#) in _open(self, file, mode_int, closefd)
   1215             err = _snd.sf_error(file_ptr)
-> 1216             raise LibsndfileError(err, prefix="Error opening {0!r}: ".format(self.name))
   1217         if mode_int == _snd.SFM_WRITE:

LibsndfileError: Error opening '/content/drive/My Drive/InputMother/WavFiles/20210907_130100.wav': System error.

During handling of the above exception, another exception occurred:

FileNotFoundError                         Traceback (most recent call last)
[/usr/local/lib/python3.10/dist-packages/opensoundscape/utils.py](https://localhost:8080/#) in make_clip_df(files, clip_duration, clip_overlap, final_clip, return_invalid_samples, raise_exceptions)
    277         try:
--> 278             t = librosa.get_duration(path=f)
    279             clips = generate_clip_times_df(

[/usr/local/lib/python3.10/dist-packages/librosa/core/audio.py](https://localhost:8080/#) in get_duration(y, sr, S, n_fft, hop_length, center, path, filename)
    803             )
--> 804             with audioread.audio_open(path) as fdesc:
    805                 return fdesc.duration  # type: ignore

[/usr/local/lib/python3.10/dist-packages/audioread/__init__.py](https://localhost:8080/#) in audio_open(path, backends)
    126         try:
--> 127             return BackendClass(path)
    128         except DecodeError:

[/usr/local/lib/python3.10/dist-packages/audioread/rawread.py](https://localhost:8080/#) in __init__(self, filename)
     58     def __init__(self, filename):
---> 59         self._fh = open(filename, 'rb')
     60 

FileNotFoundError: [Errno 2] No such file or directory: '/content/drive/My Drive/InputMother/WavFiles/20210907_130100.wav'

The above exception was the direct cause of the following exception:

GetDurationError                          Traceback (most recent call last)
[/usr/local/lib/python3.10/dist-packages/opensoundscape/annotations.py](https://localhost:8080/#) in one_hot_clip_labels(self, clip_duration, clip_overlap, min_label_overlap, min_label_fraction, full_duration, class_subset, final_clip, audio_files)
    658             try:
--> 659                 clip_df = make_clip_df(
    660                     files=[f for f in audio_files if f == f],  # remove NaN if present

[/usr/local/lib/python3.10/dist-packages/opensoundscape/utils.py](https://localhost:8080/#) in make_clip_df(files, clip_duration, clip_overlap, final_clip, return_invalid_samples, raise_exceptions)
    288             if raise_exceptions:
--> 289                 raise GetDurationError(f"Exception on file {f}") from exc
    290             else:

GetDurationError: Exception on file /content/drive/My Drive/InputMother/WavFiles/20210907_130100.wav

The above exception was the direct cause of the following exception:

GetDurationError                          Traceback (most recent call last)
[<ipython-input-105-a4d92534965b>](https://localhost:8080/#) in <cell line: 8>()
      6 
      7 # Create dataframe of one-hot labels
----> 8 clip_labels = annotations.one_hot_clip_labels(
      9     clip_duration = clip_duration,
     10     clip_overlap = clip_overlap,

[/usr/local/lib/python3.10/dist-packages/opensoundscape/annotations.py](https://localhost:8080/#) in one_hot_clip_labels(self, clip_duration, clip_overlap, min_label_overlap, min_label_fraction, full_duration, class_subset, final_clip, audio_files)
    665                 )
    666             except GetDurationError as exc:
--> 667                 raise GetDurationError(
    668                     """`full_duration` was None, but failed to retrieve the durations of 
    669                     some files. This could occur if the values of 'file' in self.df are
GetDurationError: `full_duration` was None, but failed to retrieve the durations of 
                    some files. This could occur if the values of 'file' in self.df are 
                    not paths to valid audio files. Specifying `full_duration` as an 
                    argument to `one_hot_clip_labels()` will avoid the attempt to get 
                    audio file durations from file paths."
                    
 
Deleting all the audiomoth files from the input folder, leaving just a file made by Batlogger, solves the issue.... so it cannot be a problem with file path or storage. All labels were produced in the same way with Raven, and none of the files were corrupted. 

Have searched online for solutions, including force updating soundfile. 

Providing .wav files from non-audiomoth devices works fine. Am I missing something, or is this a new issue?

**Comment by sammlapp** [comment]
Interesting, I'm not sure why audiomoth files would cause any issues compared to Batlogger files. Are you able to open those AudioMoth files on a computer? (ie, they are not corrupted or they don't have 0 length for some reason) 

You can check if CoLab can see your audiomoth file like this:
```
!ls -lh /content/drive/My Drive/InputMother/WavFiles/20210907_130100.wav
```

which should also give you the size of the file. 

If you are able to open the file and colab finds the file with the command above, please also try loading the file into an OpenSoundscape Audio object:

```python
from opensoundscape import Audio
myaudio = Audio.from_file('/content/drive/My Drive/InputMother/WavFiles/20210907_130100.wav')
myaudio
```

This error does suggest the file is not accessible or cannot be opened, so I wonder if there's something specific about these files and colab that is causing an issue. If you can share the file size of the AudioMoth files that could also be a clue. 


Finally, are you using the "triggering" feature on AudioMoths to only record based on an amplitude or frequency trigger? If so, could you share an audio file (private via email is also ok)? We don't typically use that feature, so it could be that the formatting of the AudioMoth file causes this issue.

**Comment by dsbennett2023** [comment]
I finally identified the problem, and I feel like an absolute idiot. 

My audiomoth files had the extension .WAV when they should have had .wav (lowercase). When I corrected that, they loaded correctly. 

Now, however, Im having a different issue (this time with creating annotation). 

Basically, when it reads my selection tables it is mixing up the columns - it isn't recognising my actual Annotation column as containing anything, and instead assigns a string of numbers to it. 

Im not sure what is causing this, as Ive tried importing selection tables from Raven with the default column headers ("Annotation" for the species names) and also tried with "Species" as the column name (as in your tutorial data)

It is probably, once again, an issue with the input files, because your tutorial data works perfectly.  But I cannot figure it out what is causing it or how to fix it. 

Have you heard of a similar issue?

For reference, have attached my code and some example input data. 
[Soundscape6.zip](https://github.com/kitzeslab/opensoundscape/files/13170181/Soundscape6.zip)

**Comment by sammlapp** [comment, marked as answer]
The default behavior is to guess the annotation column based on the column position (8th column), which can be buggy but at least works for the default column names in both Raven Pro and Raven Lite. If you specify the name of the column with your annotations it works properly: `BoxedAnnotations.from_raven_files(....,annotation_column_name='Annotation')`

See the documentation for details: https://opensoundscape.org/en/latest/api/modules.html#opensoundscape.annotations.BoxedAnnotations.from_raven_files
